atlas/rules/assemble.snakefile

```diff
@@ -79,19 +78,24 @@ rule quality_filter:
         config.get("threads", 1)
     resources:
         mem = config.get("java_mem", JAVA_MEM)
+    shadow: "shallow"
     shell:
-        """{SHPFXM} bbduk2.sh {params.inputs} out={output.pe} outs={output.se} \
+        """{SHPFXM} bbduk2.sh {params.inputs} out=good_quality_pe.fastq.gz outs=good_quality_se.fastq.gz \
                {params.rref} {params.lref} mink={params.mink} qout=33 \
                stats={output.stats} hdist={params.hdist} k={params.k} \
                trimq={params.trimq} qtrim={params.qtrim} threads={threads} -Xmx{resources.mem}G \
                minlength={params.minlength} minbasefrequency={params.minbasefrequency} \
-               interleaved={params.interleaved} overwrite=true 2> {log}"""
+               interleaved={params.interleaved} overwrite=true 2> {log}
+
+                cat good_quality_pe.fastq.gz good_quality_se.fastq.gz > {output.pe} 2>> {log}
```

Comment:
This has implications at the normalization and assembly steps. I would recommend implementing read joining after bbduk2.sh then concatenate joined reads, non-joined reads, and quality controlled orphans. The remainder of the protocol would then just be single-end processing.

Reply:
I don't realize what are the implication for bbnorm, but I think the single end reads should also be used to normalize the reads.